conan/cli/formatters/graph/graph.py

```diff
@@ -144,3 +144,44 @@ def format_graph_json(result):
     cli_out_write(json_result)
     if graph.error:
         raise graph.error
+
+
+def format_graph_cyclonedx(result):
+    """
+    # creates a CycloneDX JSON according to https://cyclonedx.org/docs/1.4/json/
+    """
+    def licenses(conanfilelic):
+        def entry(id):
+            return {"license": {
+                "id": id
+            }}
+        if conanfilelic is None:
+            return []
+        elif isinstance(conanfilelic, str):
+            return [entry(conanfilelic)]
+        else:
+            return [entry(i) for i in conanfilelic]
+
+    result["graph"].serialize()  # fills ids
+    deps = result["graph"].nodes[1:]  # first node is app itself
+    cyclonedx = {
+        "bomFormat": "CycloneDX",
+        "specVersion": "1.4",
+        "version": 1,
+        "dependencies": [n.id for n in deps],
```

Comment:
This code looks wrong, according to CycloneDX spec.
Could you provide an example SBOM generated with your code, so we can check if the JSON results is actually valid according to <http://cyclonedx.org/schema/bom-1.4.schema.json>

Reply:
Thanks, that is not correct. I will fix it in the conan extension